conans/client/graph/range_resolver.py

```diff
@@ -17,7 +17,12 @@ def __init__(self, conan_app, global_conf, editable_packages):
         self._resolve_prereleases = global_conf.get('core.version_ranges:resolve_prereleases')
 
     def resolve(self, require, base_conanref, remotes, update):
-        version_range = require.version_range
+        try:
+            version_range = require.version_range
+        except Exception as e:
+            base = base_conanref or "conanfile"
+            raise ConanException(f"\n    Recipe '{base}' requires '{require.ref}' "
```

Comment:
Why the extra newline at the begining?

Reply:
Because this exception is captured and re-raised by ``GraphBuilder`` and readability of the message was poor, but I didn't want to change the parent message format, because there are many tests that check it (all in a continuous line) that would break.